test/extensions/extensions_build_system.bzl

```diff
@@ -1,54 +1,55 @@
 load("//bazel:envoy_build_system.bzl", "envoy_benchmark_test", "envoy_cc_benchmark_binary", "envoy_cc_mock", "envoy_cc_test", "envoy_cc_test_binary", "envoy_cc_test_library", "envoy_py_test")
 load("@envoy_build_config//:extensions_build_config.bzl", "EXTENSIONS")
 
-def _apply_to_extension_allow_for_test(func, name, extension_name, **kwargs):
-    if not extension_name in EXTENSIONS:
-        return
+def _apply_to_extension_allow_for_test(func, name, extension_names, **kwargs):
+    for extension_name in extension_names:
+        if not extension_name in EXTENSIONS:
+            return
 
     func(name, **kwargs)
 
 # All extension tests should use this version of envoy_cc_test(). It allows compiling out
 # tests for extensions that the user does not wish to include in their build.
-# @param extension_name should match an extension listed in EXTENSIONS.
+# @param each item of extension_names should match an extension listed in EXTENSIONS.
 def envoy_extension_cc_test(
         name,
-        extension_name,
+        extension_names,
```

Comment:
I would suggest keeping extension_name to specify the primary extension that this test is for, and add an optional additional_extension_names to specify additional depedencies.  Alternatively, either have alternate extensions for tests that depend on a single extension and tests that depend on multiple extensions.  Or make extension_name optional, add an optional extension_names list and require that at least one of the two be specified for each envoy_extension_cc_test.

Part of the reason is to avoid a large scale change like this one, and also make it more clear / discoverable that some specific tests depend on multiple extensions.  Tests that depend on multiple extension should be uncommon.

Reply:
> Part of the reason is to avoid a large scale change like this one, and also make it more clear / discoverable that some specific tests depend on multiple extensions. Tests that depend on multiple extension should be uncommon.

OK that makes sense. +1 to having `make extension_name optional, add an optional extension_names list and require that at least one of the two be specified for each envoy_extension_cc_test.` since in my cases there's no primary  or secondary among these extensions.

Reply:
well on second thought, maybe I could do `additional_extension_names` as well. I want to hear others folks' preference here.

Reply:
OK added `additional_extension_names`. PTAL!

Reply:
I would suggest make `extension_name` to `extension_names`, and we can have some sort of optional extensions, but ones listed in `extension_names` is required.

Reply:
All the listed extensions are required.  I was trying to avoid having to modify all BUILD files as part of this change, but I guess we could change all the BUILD files if that's your preference.

Reply:
so I reverted back to extension_names: https://github.com/envoyproxy/envoy/pull/17181/commits/f8e22422d676e2f33cff10f9b87933ff5ad45940. anyway thanks for the suggestion @antoniovicente!